x-pack/plugins/task_manager/server/integration_tests/task_state_validation.test.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  type TestElasticsearchUtils,
+  type TestKibanaUtils,
+} from '@kbn/core-test-helpers-kbn-server';
+import { schema } from '@kbn/config-schema';
+import { TaskStatus } from '../task';
+import { type TaskPollingLifecycleOpts } from '../polling_lifecycle';
+import { type TaskClaimingOpts } from '../queries/task_claiming';
+import { TaskManagerPlugin, type TaskManagerStartContract } from '../plugin';
+import { injectTask, setupTestServers, retry } from './lib';
+
+const { TaskPollingLifecycle: TaskPollingLifecycleMock } = jest.requireMock('../polling_lifecycle');
+jest.mock('../polling_lifecycle', () => {
+  const actual = jest.requireActual('../polling_lifecycle');
+  return {
+    ...actual,
+    TaskPollingLifecycle: jest.fn().mockImplementation((opts) => {
+      return new actual.TaskPollingLifecycle(opts);
+    }),
+  };
+});
+
+const mockTaskTypeRunFn = jest.fn();
+const mockCreateTaskRunner = jest.fn();
+const mockTaskType = {
+  title: '',
+  description: '',
+  stateSchemaByVersion: {
+    1: {
+      up: (state: Record<string, unknown>) => ({ foo: state.foo || '' }),
+      schema: schema.object({
+        foo: schema.string(),
+      }),
+    },
+    2: {
```

Comment:
And if we're only allowing additive changes, is there anything we can do to enforce this? Or we'd assume the non-additive schema changes should be caught during dev time.

Reply:
>Are we only allowing additive changes to the schema so we only ever need an up migration?

Correct, for backwards compatibility, ZDT, and rollback scenarios, the approach is to only allow additive changes. The down migration in this scenario, automated by Task Manager, removes "additive fields" that an older Kibana instance doesn't understand. This is done by setting `unknowns: 'ignore'` instead of `unknowns: 'forbid'` when validating the config-schema on read.

>And if we're only allowing additive changes, is there anything we can do to enforce this? Or we'd assume the non-additive schema changes should be caught during dev time.

I'm thinking a mix of dev time and upgrade testing should catch this. We could also explore failing CI if ever we catch breaking changes (piggy back on https://github.com/elastic/kibana/issues/155784). Open to ideas!